libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI.m

```diff
@@ -272,7 +289,7 @@ - (void)enqueueRequest:(SFRestRequest *)request delegate:(id<SFRestDelegate>)del
                 [strongSelf notifyDelegateOfResponse:delegate request:request data:dataForDelegate rawResponse:response];
             }
             // 401 (and sometimes 403) indicates refresh is required.
-            else if (request.shouldRefreshOn403 ? (statusCode == 401 || statusCode == 403) : (statusCode == 401)) {
+            else if (request.requiresAuthentication && request.shouldRefreshOn403 ? (statusCode == 401 || statusCode == 403) : (statusCode == 401)) {
```

Comment:
Or maybe we should make sure shouldRetry is no if requiresAuthenticate is false.
By changing 
```objective-c
- (void)send:(SFRestRequest *)request delegate:(id<SFRestDelegate>)delegate {
    [self send:request delegate:delegate shouldRetry:YES];
}
```
to
```objective-c
- (void)send:(SFRestRequest *)request delegate:(id<SFRestDelegate>)delegate {
    if (request.requiresAuthentication) {
        [self send:request delegate:delegate shouldRetry:YES];
    }
}
```